src/crm/dynamicsWebApi.ts

```diff
@@ -59,26 +61,63 @@ const onRejected = (error: AxiosError) => {
   return Promise.reject(error);
 };
 
-const getTestStationEntities = async (requestUrl: string): Promise<DynamoTestStation[]> => {
+const getAccountsEntities = async (requestUrl: string): Promise<DynamicsTestStation[]> => {
   const siteList = (await getSecret(config.crm.siteList)).split(',');
+
+  interface AccountsFormat {
+    value: DynamicsTestStation[];
+  }
+  return lastValueFrom(
+    axios.get<AccountsFormat>(requestUrl).pipe(
+      map((data) => data.data),
+      map((data: AccountsFormat) => data.value.filter((obj) => siteList.includes(obj.dvsa_premisecodes))),
+    ),
+  );
+};
+
+const addConnectionsEmails = async (stationEntries: DynamicsTestStation[]): Promise<DynamoTestStation[]> => {
```

Comment:
This is fine but in general you want your functions to do one major thing - you're grabbing data here and also modifying a major collection. If you narrowed responsibilities then getConnectionEntities would mirror the existing getAccountsEntities function quite well. Having the caller or an additional function consume the results of that into stationEntries would give you more composable functions.